Integration tests are failing. Here's part of a stacktrace I found in the Tomcat's logs:

```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.zaxxer.hikari.HikariDataSource]: Constructor threw exception; nested exception is java.lang.ClassCastException: org.postgresql.Driver cannot be cast to javax.sql.DataSource
    at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:163)
    at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:122)
    at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:267)
    ... 46 more
Caused by: java.lang.ClassCastException: org.postgresql.Driver cannot be cast to javax.sql.DataSource
    at com.zaxxer.hikari.pool.PoolUtilities.initializeDataSource(PoolUtilities.java:102)
    at com.zaxxer.hikari.pool.BaseHikariPool.<init>(BaseHikariPool.java:156)
    at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:60)
    at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:48)
    at com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:80)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
    at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:147)
    ... 48 more
```

I suggest you use the more traditional approach by using `jdbcUrl` for configuring Hikari.

--- a/pom.xml
+++ b/pom.xml
@@ -130,6 +130,13 @@
         </dependency>
 
         <dependency>
+            <groupId>com.zaxxer</groupId>
+            <artifactId>HikariCP-java6</artifactId>
+            <version>2.3.8</version>
+            <scope>compile</scope>
+        </dependency>
+
+        <dependency>
             <groupId>org.springframework</groupId>
             <artifactId>spring-context</artifactId>
             <version>${version.spring}</version>

--- a/src/main/webapp/WEB-INF/jpa-configuration.xml
+++ b/src/main/webapp/WEB-INF/jpa-configuration.xml
@@ -50,11 +50,16 @@
         </property>
     </bean>
 
-    <bean id="dataSource" class="org.springframework.jdbc.datasource.DriverManagerDataSource">
-        <property name="driverClassName" value="${org.osiam.resource-server.db.driver}"/>
-        <property name="url" value="${org.osiam.resource-server.db.url}"/>
-        <property name="username" value="${org.osiam.resource-server.db.username}"/>
-        <property name="password" value="${org.osiam.resource-server.db.password}"/>
+    <bean id="hikariConfig" class="com.zaxxer.hikari.HikariConfig">
+        <property name="poolName" value="osiam-resource-server-cp" />
+        <property name="jdbcUrl" value="${org.osiam.resource-server.db.url}" />
+        <property name="driverClassName" value="${org.osiam.resource-server.db.driver}" />
+        <property name="username" value="${org.osiam.resource-server.db.username}" />
+        <property name="password" value="${org.osiam.resource-server.db.password}" />
+    </bean>
+
+    <bean id="dataSource" class="com.zaxxer.hikari.HikariDataSource" destroy-method="close">
+        <constructor-arg ref="hikariConfig" />
     </bean>
 
     <bean id="txManager" class="org.springframework.orm.jpa.JpaTransactionManager">

